internal/olm/operator/registry/index_image.go

```diff
@@ -100,22 +123,9 @@ func (c IndexImageCatalogCreator) UpdateCatalog(ctx context.Context, cs *v1alpha
 		}
 	}
 
-	existingItems, err := getExistingBundleItems(cs.GetAnnotations())
```

Comment:
Any reason you're moving these bits of code around? It looks like you're only adding a condition. I'd prefer minimal code changes if possible (unless you're moving this code around for a reason of course).

Reply:
The order of execution matters here. The condition sets `c.IndexImage` to `cs.Spec.Image` if no annotations exist and image reference exists (this needs to be done to support the traditional use-case). And as setting BundleAddMode is reliant on `c.IndexImage` being default or not, `BundleAddMode` should be set after the condition. 

After setting both, proceeding to creating a new annotation based on bundle image and bundle add mode, and creating annotated registry.

Reply:
Sounds good. Can you move `updateFields()` to where it was, since it only matters to `createAnnotatedRegistry()`?

Reply:
Ah, that one - yes :)

Reply:
Moved it. Also irrespective of whichever case (whether run bundle was run or if operator is installed traditionally), `cs.Spec.Image` should be set to empty, so removed the check in `updateFields`